### DISCUSSION TITLE
Run ioBroker image in a Kubernetes Cluster

### DISCUSSION THREAD

**Discussion by schoenr79** [buanet/ioBroker.docker#223] [Questions & Answers / Fragen & Antworten] [answered]
Hello community out there,
i am thinking about running the ioBroker image in an Kubernetes Cluster [K3s](https://k3s.io/) or [MicroK8s](https://microk8s.io/) on an Raspberry Pi 4. 

Does anybody tested this out already, or any objections about that idea? Im also thinking about the KNX adapter that requires host network access 

Thanks 4 your input

**Comment by dschaedl** [comment, marked as answer]
yes, that works fine. I'm running iobroker in k3s (on a raspi4 cluster) for several month now. But I don't use any special networking requirements (e.g. for homekit). I'm using Rancher to configure the iobroker deployment.

**Reply by AuqW9cQ** [reply]
Can you provide your yaml files?

That would really be incredibly helpful.

**Comment by buanet** [comment]
Seems to work, like @dschaedl said, but it's definitely not made for running in a cluster environment. As ioBroker itself is not designed to run in Docker it is not possible to do something like load balancing or high availability. So you can bring it up, but have to accept some limitations.

Regards,
André

**Comment by Apollon77** [comment]
In fact yes, simply "adding a new pod to balance" will not make any sense. But dedicated pods with own persistance that can then be restarted on issues (we would need to add the health service) or moved between nodes could work.
